hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py

```diff
@@ -10,24 +12,122 @@
 
 def lambda_handler(event, context):
     """
-    This lambda is called once per match per dataset. If a single hash matches
-    multiple datasets, this will be called multiple times.
-
-    Eventually, this will be just an action labeller. It will label a match
-    record with the action it recommends. A separate system will be stood up
-    that aggregates labels and 'decides' which action would be taken.
-
-    For now, this method will,
-    - it will construct the MatchMessage object and identify the actions it
-      needs to invoke.
-    - for now, rather than fanning out to individual specific lambdas, it will
-      call all specific functions serially!
+    This lambda is called when one or more matches are found. If a single hash matches
+    multiple datasets, this will be called only once.
+
+    Action labels are generated for each match message, then an action is performed
+    corresponding to each action label.
     """
-    for sqs_record in event["Records"]:
+    for sqs_record in event[
+        "Records"
+    ]:  # TODO research max # sqs records / lambda_handler invocation
         sns_notification = json.loads(sqs_record["body"])
         match_message: MatchMessage = MatchMessage.from_sns_message(
             sns_notification["Message"]
         )
+        action_labels = get_action_labels(match_message)
```

Comment:
I think we want to get additional labels here for reacting before we pass to the `perform_action` function. We might have a label `Label("Action", "ReactEnquedForReview") These can be calculated here and executed by `perfrom_action` as well

Reply:
@jeberl I can see how what I did with git is getting in the way here. There was a conversation that I thought came to the conclusion that we want to keep ThreatExchange-y processing separate from "core" functionality. I proposed that we have another lambda specifically for ThreatExchange reactions, hence the `ThreatExchangeReactionLabel` concept and its separate processing at 36-48.